clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -148,30 +152,38 @@ public class PrototypeAsyncConsumer<K, V> implements Consumer<K, V> {
     private boolean cachedSubscriptionHasAllFetchPositions;
     private final WakeupTrigger wakeupTrigger = new WakeupTrigger();
 
-    public PrototypeAsyncConsumer(final Properties properties,
-                                  final Deserializer<K> keyDeserializer,
-                                  final Deserializer<V> valueDeserializer) {
+    public AsyncKafkaConsumer(Map<String, Object> configs) {
```

Comment:
I really wonder if we need all those constructors now that we have the delegate creator. For instance, would it be possible to only keep the following ones and keep all the conversion to `ConsumerConfig` in the delegate creator?
```
public AsyncKafkaConsumer(final ConsumerConfig config,
                              final Deserializer<K> keyDeserializer,
                              final Deserializer<V> valueDeserializer) {
        this(Time.SYSTEM, config, keyDeserializer, valueDeserializer);
    }

    public AsyncKafkaConsumer(final Time time,
                              final ConsumerConfig config,
                              final Deserializer<K> keyDeserializer,
                              final Deserializer<V> valueDeserializer) {
```

Reply:
Yep-done.